### PR TITLE
Update Extensions.scala

### PR DESCRIPTION
**Pull request by japgolly**
* Added key codes F1~F12
* Made KeyCodes start with uppercase
* Changed `val` to `final val`

Addresses #103 and #104.

### CODE REVIEW

File: src/main/scala/org/scalajs/dom/ext/Extensions.scala [hunk 1]
```diff
@@ -96,62 +96,131 @@ object Image {
...
-  val s = 83
-  val t = 84
-  val u = 85
-  val v = 86
-  val w = 87
-  val x = 88
-  val y = 89
-  val z = 90
+  final val Backspace = 8
+  final val Tab       = 9
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/ext/Extensions.scala]: Do not align vertically. Doing it means you need to modify all lines when you add something.

File: src/main/scala/org/scalajs/dom/ext/Extensions.scala [hunk 2]
```diff
@@ -60,15 +60,15 @@ object Color {
...
-  val all = Seq(
+  final val White   = Color(255, 255, 255)
+  final val Red     = Color(255, 0, 0)
+  final val Green   = Color(0, 255, 0)
+  final val Blue    = Color(0, 0, 255)
+  final val Cyan    = Color(0, 255, 255)
+  final val Magenta = Color(255, 0, 255)
+  final val Yellow  = Color(255, 255, 0)
+  final val Black   = Color(0, 0, 0)
+  final val all = Seq(
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/ext/Extensions.scala]: There's no point for this set to be final vals. It's only meaningful for constant expressions of primitive types. So don't change these.

**Comment by sjrd** [comment]
The three bullets should be 3 commits.

**Comment by sjrd** [comment]
That's all.

**Comment by japgolly** [comment]
Feedback adhered to and resubmitted in #109

**Pull request closed without merging**